src/DotNetProjectFile.Analyzers/MsBuild/PragmaWarning.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.CodeAnalysis.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+
+namespace DotNetProjectFile.MsBuild;
+
+/// <summary>Represents a #pragma warning (disable/restore) in a MS Build project file.</summary>
+public readonly struct PragmaWarning(string diagnosticId, bool disable, Location location)
```

Comment:
I'd call this a `WarningPragma`, but I understand this matches the literal syntax better.